src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -2537,6 +2537,43 @@ def list_resource_links(cmd, scope=None, filter_string=None):
 # endregion
 
 
+# region tags
+def get_tag_at_scope(cmd, resource_id=None):
+    rcf = _resource_client_factory(cmd.cli_ctx)
+    if resource_id is not None:
+        return rcf.tags.get_at_scope(scope=resource_id)
+
+    return rcf.tags.list()
+
+
+def create_or_update_tag_at_scope(cmd, resource_id=None, tags=None, tag_name=None):
+    rcf = _resource_client_factory(cmd.cli_ctx)
+    if resource_id is not None:
+        if tags is None or bool(tags) is False:
+            raise IncorrectUsageError("Tags could not be empty.")
+        Tags = cmd.get_models('Tags')
+        tag_obj = Tags(tags=tags)
+        return rcf.tags.create_or_update_at_scope(scope=resource_id, properties=tag_obj)
+
+    return rcf.tags.create_or_update(tag_name=tag_name)
+
+
+def delete_tag_at_scope(cmd, resource_id=None, tag_name=None):
+    rcf = _resource_client_factory(cmd.cli_ctx)
+    if resource_id is not None:
+        return rcf.tags.delete_at_scope(scope=resource_id)
```

Comment:
May I ask is there need to pass in the deleted `tag_name`?

Reply:
Not in this case, if `resource_id` is not None, it would delete all the tags on this `resource_id`.

Reply:
How to delete the specified `tag_name` under the specified `resource_id`?

Reply:
At this time, this one is supported in `az tag update` with `--operation Delete`: it would delete the specific payload (we have `Replace`, `Merge` and `Delete` operations available); while `az tag delete` would delete all tags payload on the resource.